jxl/src/frame/modular/predict.rs

```diff
@@ -268,12 +268,12 @@ impl<'a> WeightedPredictorState<'a> {
         let te_w = if pos.0 == 0 {
             0
         } else {
-            self.error[cur_row + pos.0 - 1]
+            self.error[cur_row + pos.0 - 1] as i64
```

Comment:
uhm, if we actually need 64-bit arithmetic here I suspect the image might not be conformant... @jonsneyers what's your take here?

Reply:
libjxl also uses 64-bit arithmetic here:

```
    pixel_type_w teW = x == 0 ? 0 : error[cur_row + x - 1];
    pixel_type_w teN = error[pos_N];
    pixel_type_w teNW = error[pos_NW];
    pixel_type_w sumWN = teN + teW;
    pixel_type_w teNE = error[pos_NE];
```

Reply:
Also, without the as i64 it does not even compile, since all the rest of the quantities (n, w, ...) and the prediction[] array are in i64 (both here and in libjxl).